### PR TITLE
feat: automate ocr2vrf job creation

### PR DESCRIPTION
**Pull request by makramkd**
Tweak existing automation code to create OCR2VRF jobs as well.

### CODE REVIEW

File: core/scripts/ocr2vrf/setup_dkg.go [hunk 1]
```diff
@@ -155,7 +156,7 @@ func resetDatabase(client *cmd.Client, context *cli.Context, index int, database
 	helpers.PanicErr(client.ResetDatabase(context))
 }
 
-func newDKGSetupClient() *cmd.Client {
+func newSetupClient() *cmd.Client {
```
Comment (vreff) [inline core/scripts/ocr2vrf/setup_dkg.go:159]: `newSetupClient` and `setupOCR2VRFNodeFromClient` might be better in the util.go file now that they're being used by multiple files.

File: core/scripts/ocr2vrf/setup_ocr2vrf.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+func setupOCR2VRFNodes(e helpers.Environment) {
+	client := newSetupClient()
+	app := cmd.NewApp(client)
+
+	fs := flag.NewFlagSet("ocr2vrf-setup", flag.ExitOnError)
+
+	keyID := fs.String("key-id", "aee00d81f822f882b6fe28489822f59ebb21ea95c0ae21d9f67c0239461148fc", "key ID")
+	linkAddress := fs.String("link-address", "", "LINK token address")
+	linkEthFeed := fs.String("link-eth-feed", "", "LINK-ETH feed address")
+	confDelays := fs.String("conf-delays", "1,2,3,4,5,6,7,8", "8 confirmation delays")
```
Comment (vreff) [inline core/scripts/ocr2vrf/setup_ocr2vrf.go:35]: always 8? If so should we enforce that?

File: core/scripts/ocr2vrf/setup_ocr2vrf.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		onChainPublicKeys = append(onChainPublicKeys, payload.OnChainPublicKey)
+		offChainPublicKeys = append(offChainPublicKeys, payload.OffChainPublicKey)
+		configPublicKeys = append(configPublicKeys, payload.ConfigPublicKey)
+		peerIDs = append(peerIDs, payload.PeerID)
+		transmitters = append(transmitters, payload.Transmitter)
+		dkgEncrypters = append(dkgEncrypters, payload.DkgEncrypt)
+		dkgSigners = append(dkgSigners, payload.DkgSign)
+	}
+
+	fundNodes(e, transmitters, *fundingAmount)
```
Comment (vreff) [inline core/scripts/ocr2vrf/setup_ocr2vrf.go:129]: might also be a util.go function.

File: core/scripts/ocr2vrf/util.go [hunk 4]
```diff
@@ -311,3 +312,11 @@ func decodeHexTo32ByteArray(val string) (byteArray [32]byte) {
...
 	return
 }
+
+func deployLinkEthFeed(e helpers.Environment, linkAddress string, weiPerUnitLink *big.Int) common.Address {
+	_, tx, _, err :=
+		mock_v3_aggregator_contract.DeployMockV3AggregatorContract(
+			e.Owner, e.Ec, 18, weiPerUnitLink)
+	helpers.PanicErr(err)
+	return helpers.ConfirmContractDeployed(context.Background(), e.Ec, tx, e.ChainID)
+}
```
Comment (vreff) [inline core/scripts/ocr2vrf/util.go:322]: noice

File: core/scripts/ocr2vrf/setup_ocr2vrf.go (hunk 2 above)
Comment (makramkd) [reply core/scripts/ocr2vrf/setup_ocr2vrf.go:35]: Enforcing now.

File: core/scripts/ocr2vrf/setup_ocr2vrf.go (hunk 3 above)
Comment (makramkd) [reply core/scripts/ocr2vrf/setup_ocr2vrf.go:148]: Used the one in helpers.

File: core/scripts/ocr2vrf/util.go (hunk 4 above)
Comment (makramkd) [reply core/scripts/ocr2vrf/util.go]: Removed, using the one in helpers instead.

File: core/scripts/ocr2vrf/setup_dkg.go (hunk 1 above)
Comment (makramkd) [reply core/scripts/ocr2vrf/setup_dkg.go]: Moved this and the other functions that are common between the two.

File: core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go [hunk 5]
```diff
@@ -407,6 +429,11 @@ func (c *coordinator) unmarshalLogs(
 				return
 			}
 			newTransmissionLogs = append(newTransmissionLogs, unpacked)
+		default:
+			c.lggr.Info(fmt.Sprintf("Unexpected event sig: %s", hexutil.Encode(lg.EventSig)))
+			c.lggr.Info(fmt.Sprintf("expected one of: %s %s %s %s",
```
Comment (jinhoonbang) [inline core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go:434]: Should these be warning or error logs?

**Comment by vreff** [comment]
LGTM. We don't get the cozy "NEW KEY FOR KEY ID" log on the DKG anymore in data availability phase with this setup, but that's just an observation not a blocker. Can approve pending Jin's comment

File: core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go (hunk 5 above)
Comment (makramkd) [reply core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go:434]: Good note, I think these would be better served as errors.

**Review by vreff — APPROVED** [review]

**Review by jinhoonbang — APPROVED** [review]

**Pull request merged**